layouts/partials/kubectl-reference-index.html

```diff
@@ -0,0 +1,48 @@
+{{- $parent := .Page -}}
+{{- $pages := (where .Site.Pages "Section" .Section).ByWeight -}}
+{{- $pages = (where $pages "Type" "!=" "search") -}}
+{{- $pages = (where $pages ".Parent" "!=" nil) -}}
+{{- $pages = (where $pages "Parent.File.UniqueID" "==" $parent.File.UniqueID) -}}
+
+<hr class="panel-line">
+<div id="kubectl-reference-index">
+
+    {{- range $index, $childPage := $pages -}} <!-- Iterate through all the section pages -->
+        {{- $sectionId := (replace $childPage.Title " " "-") | lower -}} <!-- Generate unique ID for each section page -->
```

Comment:
Does this handle duplicates? Hugo won't cover that for us.

Reply:
Currently, duplicates aren't being considered in this implementation. The loop just iterates over all the pages under "[docs/reference/kubectl/generated(here)](https://github.com/kubernetes/website/tree/main/content/en/docs/reference/kubectl/generated)"  directory here to render them.